lib/types.nix

```diff
@@ -32,7 +39,17 @@ rec {
     check = v: (isRawType v) || (v ? __empty);
   };
 
-  maybeRaw = type: types.either type rawLua;
+  maybeRaw =
+    type:
+    let
+      luaFirst = types.either rawLua type;
+      typeFirst = types.either type rawLua;
+    in
+    luaFirst
+    // {
+      name = "maybeRaw";
+      inherit (typeFirst) description;
+    };
```

Comment:
@traxys are you still happy after this addition to the PR?
This fixes https://github.com/nix-community/nixvim/pull/2337#issuecomment-2381400788.

Reply:
https://github.com/nix-community/nixvim/pull/2337#issuecomment-2381405395 was resolved by also overriding `nestedTypes` to match the prior attrs.